src/vs/workbench/services/notification/common/notificationService.ts

```diff
@@ -119,12 +124,22 @@ export class NotificationService extends Disposable implements INotificationServ
 			// If the user already picked to not show the notification
 			// again, we return with a no-op notification here
 			if (this.storageService.getBoolean(id, scope)) {
+				// check if key already in storage
+				const notifications = this.getNotifications();
+				notifications.push({ neverShowAgain: true, id: id, label: 'label', when: 'whenever' });
+				this.storageService.store('dismissedNotifications', JSON.stringify(notifications), scope, StorageTarget.USER);
+
 				return new NoOpNotification();
 			}
 
 			const neverShowAgainChoice = {
 				label: nls.localize('neverShowAgain', "Don't Show Again"),
-				run: () => this.storageService.store(id, true, scope, StorageTarget.USER),
+				run: () => {
+					this.storageService.store(id, true, scope, StorageTarget.USER);
+
+					this.getNotifications().push(<INotificationItem>{ neverShowAgain: true, id: id, label: 'label', when: 'whenever' });
```

Comment:
as we discussed, this won't work because we can't get all the info we need from the notification here. will have to register it with an id/key, label, and when description of when the notification gets fired